frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,390 @@
+  | "naming.index.uk"
+  | "naming.index.idx"
+  | "column.required"
+  | "column.no-null"
+  | "query.select.no-select-all"
+  | "query.where.require"
+  | "query.where.no-leading-wildcard-like";
+
+export interface RuleTemplate {
+  id: RuleId;
+  category: CategoryType;
+  engine: SchemaRuleEngineType;
+  description: string;
+  payload?: RulePayload;
+  level: RuleLevel;
+}
+
+interface BaseSchemaRuleType {
+  id: RuleId;
+  level: RuleLevel;
+}
+
+interface NamingFormatRuleType extends BaseSchemaRuleType {
+  format: string;
+}
+
+interface RequiredColumnRuleType extends BaseSchemaRuleType {
+  columnList: string[];
+}
+
+type SchemaPolicyRule =
+  | BaseSchemaRuleType
+  | NamingFormatRuleType
+  | RequiredColumnRuleType;
+
+export interface DatabaseSchemaReviewPolicy {
+  id: SchemaReviewPolicyId;
+
+  // Standard fields
+  creator: Principal;
+  createdTs: number;
+  updater: Principal;
+  updatedTs: number;
+
+  // Domain specific fields
+  name: string;
+  ruleList: SchemaPolicyRule[];
+  environmentIdList: number[];
+}
+
+export interface DatabaseSchemaReviewPolicyCreate {
+  // Domain specific fields
+  name: string;
+  ruleList: SchemaPolicyRule[];
+  environmentIdList: number[];
+}
+
+export type DatabaseSchemaReviewPolicyPatch = {
+  // Domain specific fields
+  name?: string;
+  ruleList?: SchemaPolicyRule[];
+  environmentIdList?: number[];
+};
+
+interface RuleCategory {
+  id: CategoryType;
+  ruleList: RuleTemplate[];
+}
+
+export interface SchemaReviewTemplate {
+  name: string;
+  imagePath: string;
+  ruleList: RuleTemplate[];
+}
+
+const categoryOrder: Map<CategoryType, number> = new Map([
+  ["ENGINE", 5],
+  ["NAMING", 4],
+  ["QUERY", 3],
+  ["TABLE", 2],
+  ["COLUMN", 1],
+]);
+
+export const convertToCategoryList = (
+  ruleList: RuleTemplate[]
+): RuleCategory[] => {
+  const dict = ruleList.reduce((dict, rule) => {
+    if (!dict[rule.category]) {
+      const id = rule.category.toLowerCase();
+      dict[rule.category] = {
+        id: rule.category,
+        ruleList: [],
+      };
+    }
+    dict[rule.category].ruleList.push(rule);
+    return dict;
+  }, {} as { [key: string]: RuleCategory });
+
+  return Object.values(dict).sort(
+    (c1, c2) =>
+      (categoryOrder.get(c2.id as CategoryType) || 0) -
+      (categoryOrder.get(c1.id as CategoryType) || 0)
+  );
+};
+
+// TODO: i18n
+export const ruleList: RuleTemplate[] = [
+  {
+    id: "engine.mysql.use-innodb",
+    category: "ENGINE",
+    engine: "MYSQL",
+    description: "Require InnoDB as the storage engine.",
+    level: RuleLevel.ERROR,
+  },
+  {
+    id: "table.require-pk",
+    category: "TABLE",
+    engine: "COMMON",
+    description: "Require the table to have a primary key.",
+    level: RuleLevel.ERROR,
+  },
+  {
+    id: "naming.table",
+    category: "NAMING",
+    engine: "COMMON",
+    description: "Enforce the table name format. Default snake_lower_case.",
+    level: RuleLevel.ERROR,
+
+    payload: {
+      format: {
```

Comment:
Since we know the format from the id, thus the field format/type field seems to be unnecessary.

payload: {
defualt: "^[a-z]+(_[a-z]+)?$"
}

Similar for other types.

Reply:
Don't we support users changing the rule configuration anymore?

Reply:
Seems I mistake. But if we don't support custom configuration for this rule, seems we should rename it as "naming.table.snake-lower-case"?